.travis.yml

```diff
@@ -23,14 +23,36 @@ matrix:
         - ghc-8.6.1
         - cabal-install-head
 
-install:
-- export PATH="/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$PATH"
-- echo $PATH
-- cabal new-update
-- cabal new-build --enable-tests --enable-benchmarks --allow-newer
+ - ghc: 8.6.1
+    env: GHCVER='8.6.1' STACK_YAML="$TRAVIS_BUILD_DIR/stack-$GHCVER.yaml"
+    os: linux
+    addons:
+      apt:
+        packages:
+        - libgmp-dev
 
+install:
+  - |
+    if [ -z "$STACK_YAML" ]; then
+      export PATH="/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$PATH"
+      echo $PATH
+      cabal new-update
+      cabal new-build --enable-tests --enable-benchmarks --allow-newer
+    else
+      mkdir -p ~/.local/bin
+      export PATH="$HOME/.local/bin:$PATH"
+      travis_retry curl -L 'https://www.stackage.org/stack/linux-x86_64' | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+      stack --version
+      stack setup --no-terminal --install-cabal 2.2.0.1
+      stack ghc -- --version
+      stack build --only-dependencies --no-terminal
+    fi
 script:
-- cabal new-test --allow-newer
-
+  - |
+    if [ -z "$STACK_YAML" ]; then
+       cabal new-test --allow-newer
```

Comment:
@v0d1ch Could you try to remove `--allow-newer` here? It shouldn't be needed anyway. And this change also will trigger Travis CI rebuild. Somehow the report is stucked...

Reply:
I removed it from two places, let's see if it builds on travis now.